lib/gtfs.ex

```diff
@@ -133,34 +134,76 @@ defmodule Gtfs do
   def handle_continue({:load_gtfs, files_source, health_server_pid}, :not_loaded) do
     start_time = Time.utc_now()
 
-    case fetch_files(files_source) do
+    with {:ok, data} <- fetch_gtfs(files_source) do
+      state = {:loaded, data}
+
+      Logger.info(fn ->
+        "Successfully loaded gtfs, time_in_ms=#{
+          Time.diff(Time.utc_now(), start_time, :millisecond)
+        }"
+      end)
+
+      if health_server_pid do
+        HealthServer.loaded(health_server_pid)
+      end
+
+      {:noreply, state}
+    else
       {:error, error} ->
         Logger.info(fn ->
           "Error loading gtfs, time_in_ms=#{Time.diff(Time.utc_now(), start_time, :millisecond)}"
         end)
 
         {:stop, error}
+    end
+  end
 
-      files ->
-        data = parse_files(files)
-        state = {:loaded, data}
+  @spec fetch_gtfs(files_source()) :: {:ok, t()} | {:error, any()}
+  defp fetch_gtfs({:mocked_files, mocked_files}) do
+    data =
+      mocked_files
+      |> files_from_mocked()
+      |> parse_files()
 
-        Logger.info(fn ->
-          "Successfully loaded gtfs, time_in_ms=#{
-            Time.diff(Time.utc_now(), start_time, :millisecond)
-          }"
-        end)
+    {:ok, data}
+  end
+
+  defp fetch_gtfs({:url, url}) do
+    if CacheFile.should_use_file?() do
+      Logger.info("Loading gfts data from cached file")
+
+      with {:ok, data} <- CacheFile.load_gtfs() do
+        {:ok, data}
+      else
+        _ ->
+          with {:ok, data} <- fetch_and_parse_data(url) do
+            CacheFile.save_gtfs(data)
+            {:ok, data}
+          else
+            {:error, error} ->
+              {:error, error}
+          end
+      end
+    else
+      fetch_and_parse_data(url)
+    end
+  end
 
-        if health_server_pid do
-          HealthServer.loaded(health_server_pid)
-        end
+  @spec fetch_and_parse_data(String.t()) :: {:ok, t()} | {:error, any()}
```

Comment:
There are a bunch of functions named `fetch_` here:
`files_from_mocked`, `fetch_files`, `fetch_and_parse_data`, `fetch_gtfs`.
The names don't make it clear which ones might hit the filesystem and which ones might hit the internet.
What do you think about naming them:
`files_from_mocked`, `files_from_url`, `gtfs_from_url`, and `fetch_gtfs`?